addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/BaseKeypadHandler.java

```diff
@@ -195,12 +196,39 @@ private void asyncInitialize() {
 
             configureChannels();
 
-            updateStatus(ThingStatus.ONLINE);
+            initDeviceState();
 
             logger.debug("Async init thread finishing for keypad handler {}", integrationId);
         }
     }
 
+    @Override
+    public void initDeviceState() {
+        synchronized (asyncInitLock) {
+            logger.debug("Initializing device state for Keypad {}", integrationId);
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+            } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+                if (ledList.isEmpty()) {
+                    // Device with no LEDs has nothing to query. Assume it is online if bridge is online.
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    // Query LED states. Method handleUpdate() will set thing status to online when response arrives.
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Awaiting initial response");
+                    // To reduce query volume, query only 1st LED and LEDs with linked channels.
+                    for (KeypadComponent component : ledList) {
+                        if (component.id() == ledList.get(0).id() || isLinked(channelFromComponent(component.id()))) {
+                            queryDevice(component.id(), ACTION_LED_STATE);
```

Comment:
I guess we can break the `for` loop after the first match. Can we?

Reply:
No, we needs to iterate through all of them so that any button LED with a linked channel can be queried. Their states need to be updated when the bridge status transitions back to ONLINE from OFFLINE, since they may be stale.  I query the first one just in case no LED channels are linked, since we need to generate at least one query and get a response to finish updating the device state.

Reply:
:thumbsup: